utils/stall_free.hh

```diff
@@ -56,14 +58,99 @@ void merge_to_gently(std::list<T>& list1, const std::list<T>& list2, Compare com
     }
 }
 
-template<class T>
-seastar::future<> clear_gently(std::list<T>& list) {
-    return repeat([&list] () mutable {
-        if (list.empty()) {
-            return seastar::stop_iteration::yes;
-        }
-        list.pop_back();
-        return seastar::stop_iteration::no;
+template <typename T>
+concept GentlyClearable = requires (T x) {
+    { x.clear_gently() } -> std::same_as<future<>>;
+};
+
+template <typename Ptr, typename T = typename Ptr::element_type>
+concept SmartPointer = requires (Ptr x) {
```

Comment:
You are mixing SFINAE-style programming with concepts. No need to define T here, just add a requirement `typename Ptr::element_type` below.

Reply:
Fixes in v5